authors/apps/bookmark/views.py

```diff
@@ -0,0 +1,81 @@
+from rest_framework import generics, status
+from rest_framework.response import Response
+from rest_framework.permissions import (IsAuthenticatedOrReadOnly,
+                                        IsAuthenticated)
+from django.core.exceptions import ObjectDoesNotExist
+from django.db.utils import IntegrityError
+from ..articles.models import Article
+
+from .serializers import BookmarksSerializer
+from .models import Bookmarks
+
+
+class CreateBookmark(generics.CreateAPIView):
+    serializer_class = BookmarksSerializer
+    queryset = Bookmarks.objects.all()
+    permission_classes = (IsAuthenticatedOrReadOnly,)
+
+    def create(self, request, slug, *args, **kwargs):
+        user = request.user
+        try:
+            article_instance = Article.objects.get(slug=slug)
+        except ObjectDoesNotExist:
+            return Response(
+                {'error': 'Article {} does not exist'.format(slug)},
+                status.HTTP_404_NOT_FOUND)
+        data = {
+            "article": article_instance.id,
+            "user": user.id}
+        serializer = self.serializer_class(data=data)
+        try:
+            serializer.is_valid(raise_exception=True)
+            serializer.save()
+        except IntegrityError:
+            return Response(
+                {'error': 'Article already exist in your bookmark'},
+                status.HTTP_400_BAD_REQUEST)
+        return Response(
+            {'message': 'Article  has been added to your bookmark'},
+            status.HTTP_201_CREATED)
+
+
+class ListALlBookmarks(generics.ListAPIView):
+    serializer_class = BookmarksSerializer
+    queryset = Bookmarks.objects.all()
+    permission_classes = (IsAuthenticated,)
+
+    def list(self, request, *args, **kwargs):
+        bookmark = Bookmarks.objects.filter(user=request.user)
+        serializer = self.serializer_class(bookmark, many=True)
+        data = serializer.data
+        if len(data) == 0:
+            return Response(
+                {'error': 'You have no bookmark available'},
```

Comment:
That a user doesn't have bookmarks is not really an error. Let's return an empty array instead